src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -727,20 +728,31 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
                         // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        Vector128<sbyte> sourceValue;
+                        bool allBytesAreAscii;
 
                         // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
                         // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        if (Sse2.IsSupported)
+                        {
+                            sourceValue = Sse2.LoadVector128(startingAddress);
+                            allBytesAreAscii = Sse2.MoveMask(sourceValue) == 0;
+                        }
+                        else
+                        {
+                            Debug.Assert(AdvSimd.Arm64.IsSupported);
+                            sourceValue = AdvSimd.LoadVector128(startingAddress);
+                            allBytesAreAscii = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() >= 0;
```

Comment:
Same question as above - what made you decide to use `AdvSimd.Arm64.MinAcross(sourceValue).ToScalar()` instead of your new `AdvSimdHelper.MoveMask` method?

There is also another case in the next file. If the answer is the same as in the first case, ignore this repeated question. 😄